### DISCUSSION TITLE
Improve cache handling for local development

### DISCUSSION THREAD

**Discussion by itpropro** [nitrojs/nitro#598] [Ideas]
If using e.g. `defineCachedFunction`, the `.nitro/cache/nitro_` folder stays on the fs, even if the dev server is shut down. For a better local dev experience, the cache should either be cleared at shut down or newly initialized at start or alternatively provide a cli switch like `--clearCache` to provide full flexibility to persist the cache, if needed.

**Comment by itpropro** [comment]
I can try to come up with a PR for the CLI parameter, I think that is the most flexible solution, what do you think @danielroe ?

**Comment by pi0** [comment]
Hi @itpropro. You can use `devStorage: { cache: { driver: 'memory' } }` in order to override it. Using fs was purposefully to keep cache across dev reload/restart. We can however support an option like `cleanCache` for nitro to make the behavior you want. PR welcome :)

**Reply by itpropro** [reply]
Hey,
I'm normally using `driver: 'memory'`, which is why I have not recognized the persistence of the json files so far. I was just looking for a way to choose between a clean startup or cache persistence in some transparent way. If you think the most flexible way for nitro users is with a CLI switch (or maybe a config setting?), I will try to make a PR for that :)

**Reply by pi0** [reply]
Sure we can have both option and switch to override option 👍🏼
